### ISSUE TITLE
Bug

### ISSUE THREAD

**Issue by Wjxfi** [deltazefiro/Amarok-Hider#25]
I used Amarok on the usual Pictures folder and Amarok did nothing. Just the hide button turned gray

miui 12.0.5
android 12
ver. 0.8.2b4-foss

more info:
There were a lot of folders but not many pictures, no more than 200

**Comment by deltazefiro** [comment]
Thanks for the report. It seems that Amarok was running properly - It takes a while to rename so many folders - and there should be a spinning process indicator to the right of the hide button. 
If you are sure it really "did nothing", can you provide me with the logs? You can get them by logcat!

**Comment by deltazefiro** [comment]
> You absolutely need to increase the performance of the application. For one folder it's good and for several folders it's terrible

:disappointed: I've tried my best. Amarok does well in dealing with large files, but not lots of small files.  
I also did some profiling. It shows the bottleneck is the file renaming.  
If you get some awesome ideas to improve the performance, feel free to open an FR issue.  

btw, I'm working on turning Amarok into an Xposed module. It can literally hide the file in instant, but a rooted device is a must.
